integration/spark/src/main/scala/org/apache/spark/sql/secondaryindex/rdd/SecondaryIndexCreator.scala

```diff
@@ -398,6 +404,22 @@ object SecondaryIndexCreator {
           secondaryIndexModel.sqlContext.sparkSession,
           carbonLoadModelForMergeDataFiles.getFactTimeStamp,
           rebuiltSegments)
+
+        if (isInsertOverwrite) {
+          FileInternalUtil
+            .updateTableStatus(
+              SegmentStatusManager
+                .readLoadMetadata(indexCarbonTable.getMetadataPath)
+                .filter(loadMetadata => !successSISegments.contains(loadMetadata.getLoadName))
+                .map(_.getLoadName).toList,
+              secondaryIndexModel.carbonLoadModel.getDatabaseName,
```

Comment:
please do not add complete filter logic inside a method parameter, please assign that to a variable for clean code practice.

Reply:
done